src/Cli/dotnet/SlnFileFactory.cs

```diff
@@ -126,5 +125,37 @@ public static SolutionModel CreateFromFilteredSolutionFile(string filteredSoluti
 
             return filteredSolution;
         }
+
+        public static string GetSolutionPathFromFilteredSolutionFile(string filteredSolutionPath)
+        {
+            try
+            {
+                JsonElement jsonElement = ParseSolutionFilterFile(filteredSolutionPath);
+                JsonElement filteredSolutionJsonElement = jsonElement.GetProperty("solution");
+                string originalSolutionPath = filteredSolutionJsonElement.GetProperty("path").GetString();
+                return originalSolutionPath;
+            }
+            catch (Exception ex)
+            {
+                throw new GracefulException(
+                    CommonLocalizableStrings.InvalidSolutionFormatString,
+                    filteredSolutionPath, ex.Message);
+            }
+        }
+
+        private static JsonElement ParseSolutionFilterFile(string solutionFilterFilePath)
```

Comment:
Maybe 
```suggestion
        private static JsonElement ParseSolutionFilterFileJson(string solutionFilterFilePath)
```
just to make it super clear that it returns a json?

Reply:
For now, I removed it as it's only used by 1 method.